platform/ext/target/nordic_nrf/common/nrf5340/tfm_hal_its_encryption.c

```diff
@@ -0,0 +1,204 @@
+#include "platform/include/tfm_platform_system.h"
+#include "platform/include/tfm_hal_its.h"
+#include "psa/crypto.h"
+
+#include "nrf_cc3xx_platform_derived_key.h"
+#include "nrf_cc3xx_platform.h"
+/* TODO: Remove this after testing */
+#include "nrf_cc3xx_platform_kmu.h"
+#include <string.h>
+
+#define ITS_ENCRYPTION_SUCCESS 0
+
+#define HUK_KMU_SLOT 2
+#define HUK_KMU_SIZE_BITS 256
+
+static int32_t tfm_hal_its_aead_crypt(uint8_t *plaintext,
+                                      size_t plaintext_size,
+                                      uint8_t *ciphertext,
+                                      size_t ciphertext_size,
+                                      uint8_t *deriv_label,
+                                      size_t deriv_label_size,
+                                      uint8_t *nonce,
+                                      size_t nonce_size,
+                                      uint8_t *aad,
+                                      size_t aad_size,
+                                      uint8_t *tag,
+                                      size_t tag_size,
+                                      bool isEncrypt)
+{
+    int err = NRF_CC3XX_PLATFORM_ERROR_INTERNAL;
+    nrf_cc3xx_platform_derived_key_ctx_t deriv_ctx = {0};
+
+    err = nrf_cc3xx_platform_derived_key_init(&deriv_ctx);
+    if (err != NRF_CC3XX_PLATFORM_SUCCESS){
+        return PSA_ERROR_HARDWARE_FAILURE;
+    }
+
+    err = nrf_cc3xx_platform_derived_key_set_info(&deriv_ctx,
+                                                  HUK_KMU_SLOT,
+                                                  HUK_KMU_SIZE_BITS,
+                                                  deriv_label,
+                                                  deriv_label_size);
+    if (err != NRF_CC3XX_PLATFORM_SUCCESS){
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    err = nrf_cc3xx_platform_derived_key_set_cipher(&deriv_ctx,
+                                                    ALG_CHACHAPOLY_256_BIT);
+
+    if (err != NRF_CC3XX_PLATFORM_SUCCESS){
+        return PSA_ERROR_HARDWARE_FAILURE;
+    }
+
+    err = nrf_cc3xx_platform_derived_key_set_auth_info(&deriv_ctx,
+                                                       nonce,
+                                                       nonce_size,
+                                                       aad,
+                                                       aad_size,
+                                                       tag,
+                                                       tag_size);
+    if (err != NRF_CC3XX_PLATFORM_SUCCESS){
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+
+    if(isEncrypt){
+        err = nrf_cc3xx_platform_derived_key_encrypt(&deriv_ctx,
+                                                     ciphertext,
+                                                     plaintext_size,
+                                                     plaintext);
+
+    } else
```

Comment:
The style on the if / else in this file need to be fixed at some point.

Reply:
Should be fixed now I think